Rebus.AzureQueues/Config/AzureStorageQueuesConfigurationExtensions.cs

```diff
@@ -63,6 +64,24 @@ public static void UseAzureStorageQueuesAsOneWayClient(this StandardConfigurer<I
             OneWayClientBackdoor.ConfigureOneWayClient(configurer);
         }
 
+        /// <summary>
+        /// Configures Rebus to use Azure Storage Queues to transport messages
+        /// </summary>
+        public static void UseAzureStorageQueues(this StandardConfigurer<ITransport> configurer, ICloudQueueFactory queueFactory, string inputQueueAddress, AzureStorageQueuesTransportOptions options = null)
+        {
+            Register(configurer, inputQueueAddress, queueFactory, options);
+        }
+
+        /// <summary>
+        /// Configures Rebus to use Azure Storage Queues to transport messages as a one-way client (i.e. will not be able to receive any messages)
+        /// </summary>
+        public static void UseAzureStorageQueuesAsOneWayClient(this StandardConfigurer<ITransport> configurer, ICloudQueueFactory queueFactory, AzureStorageQueuesTransportOptions options = null)
+        {
+            Register(configurer, null, queueFactory, options);
+
+            OneWayClientBackdoor.ConfigureOneWayClient(configurer);
+        }
```

Comment:
these two configuration overloads would not be necessary, if the `ICloudQueueFactory` got resolved from Injectionis

Reply:
How would one configure AzureStorageQueues then? The other `UseAzureStorageQueues` extensions require either a `CloudStorageAccount` or a `string` connection string, neither of which is applicable when you want to use a custom `ICloudQueueFactory`

Reply:
hmmm good point.... you are probably right that, since this particular extension actually voids the need for `CloudStorageAccount`/connection string, then it's appropriate to add the methods as overloads